packages/components/modules/messages/ChatRoomsList/ChatRoomItem/index.tsx

```diff
@@ -53,16 +57,43 @@ const ChatRoomItem: FC<ChatRoomItemProps> = ({
 
   const showBadge = room.unreadMessagesCount && room.unreadMessagesCount > 0
 
+  const [commit, isMutationInFlight] = useArchiveChatRoomMutation()
+
   return (
     <ActionsOverlay
       title="Chat"
       offsetTop={-12}
       actions={[
         {
-          disabled: true,
+          disabled: isMutationInFlight,
           icon: <ArchiveIcon />,
           label: 'Archive Chat',
-          onClick: () => {},
+          onClick: () => {
+            if (currentProfile?.id) {
+              commit({
+                variables: {
+                  input: {
+                    roomId: room.id,
+                    profileId: currentProfile.id,
+                    archive: !isInArchivedTab,
+                  },
+                },
+                updater: (store: RecordSourceSelectorProxy<unknown>) => {
```

Comment:
This does not depend on the server response. So it always removes the chat, even if there was an error on the backend? Or am I missing something?

Reply:
If I'm not mistaken, relay doesn't update if there is a server error. But I don't think it would hurt to include error handling in that function, to be safe